library/core/src/ops/control_flow.rs

```diff
@@ -0,0 +1,67 @@
+use crate::ops::Try;
+
+/// Used to make try_fold closures more like normal loops
+#[unstable(feature="control_flow_enum", reason="new API", issue="75744")]
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum ControlFlow<C, B> {
```

Comment:
I have found that `C = ()` is very common when using this enum, common enough that it might deserve a defaulted type parameter. We could add one backwards compatibly but only if `C` follows `B` in the parameter list. Otherwise we would have to default `B` at the same time and anyone that wishes to override it would have to override `C` as well.

Reply:
Good point!  It's even usually that way in the uses this PR touches, like
![image](https://user-images.githubusercontent.com/18526288/91897342-a3bd7a80-ec89-11ea-9fb9-dee05328fbda.png)
I suppose we could even do `ControlFlow<B = (), C = ()>` the same way `break;` is `break ();`.

(Though I suppose the reorder might make the diff messier, so could also be a separate PR.)

Reply:
I think it would be confusing to do this without changing the order of the variants as well, and we may as well do that too.

Reply:
On consideration, I think a separate PR would be best.